### DISCUSSION TITLE
Close stashable sidebar on Esc key

### DISCUSSION THREAD

**Discussion by jeffchown** [livewire/flux#594] [Feature Requests]
I think it would add consistency (with flyout modals) and increased ease of use to make it possible to close `stashable` `flux:sidebar`s via the Esc key.

Maybe via a new bool attribute (e.g. `close-on-esc="true"`) in the `<flux:sidebar close-on-esc="true">...</flux:sidebar>` component

**Comment by calebporzio** [comment]
Hmmm - stashable sidebars are really meant for mobile (which users don't typically use a keyboard for) - in what context are you running into this and want this?

**Reply by jeffchown** [reply]
@calebporzio Agreed that they are primarily for mobile.  In this case, I have an app that has three very complicated forms (medical information) for which I provide the user with the ability to toggle a full screen 'focus' mode to make the most of the screen's real estate.

When in full screen mode, they access the stashable sidebar menu and different flyout modals (for different form-related functions) and can close the flyout modals with Esc but can't close the 'flyout' sidebar menu the same way.  This makes the UX a little inconsistent and forces them to click the X to close the sidebar.

While this is a specific use-case, this is not a one-off as I do sometimes like to offer users the ability to stash the sidebar on desktop to enable them to make the most of the width of their screen (especially if they are on a small laptop).
